common/app/common/InlineStyles.scala

```diff
@@ -93,21 +94,48 @@ object InlineStyles {
   }
 
   /**
+    * Execute some code n times, or until no exception is thrown
+    * (If n is 0, code still gets executed once)
+    */
+  def retry[T](n: Int)(r: => T)(implicit onFail: (Throwable, Int) => Unit) = {
+    def go(i: Int): Try[T] = {
+      Try(r) match {
+        case Failure(e) if i < n => {
+          onFail(e, i)
+          go(i + 1)
+        }
+        case f@Failure(e) => {
+          onFail(e, i)
+          f
+        }
+        case s@Success(v) => s
+      }
+    }
+
+    go(1)
+  }
+
+  /**
     * Convert the styles in a document's <style> tags to a pair.
     * The first item is the styles that should stay in the head, the second is everything that should be inlined.
     */
   def styles(document: Document): (Seq[CSSRule], Seq[String]) = {
     document.getElementsByTag("style").foldLeft((Seq.empty[CSSRule], Seq.empty[String])) { case ((inline, head), element) =>
       val source = new InputSource(new StringReader(element.html))
 
-      Try(cssParser.parseStyleSheet(source, null, null)).toOption map { sheet =>
-        val (styles, others) = seq(sheet.getCssRules).partition(isStyleRule)
-        val (inlineStyles, headStyles) = styles.flatMap(CSSRule.fromW3).flatten.partition(_.canInline)
-        val newHead = (headStyles.map(_.toString) ++ others.map(_.getCssText)).mkString("\n")
-
-        (inline ++ inlineStyles, (head :+ newHead).filter(_.nonEmpty))
-      } getOrElse {
-        (inline, head :+ element.html)
+      retry(3)(cssParser.parseStyleSheet(source, null, null)) { (e: Throwable, i: Int) =>
```

Comment:
Throwable and Int should be inferred for you here.  Longer variable names might be more helpful than types in this case.